x-pack/plugins/runtime_fields/README.md

```diff
@@ -61,7 +62,40 @@ const MyComponent = () => {
 }
 ```
 
-#### Alternative
+#### `@returns`
+
+You get back a `RuntimeField` object with the following interface
+
+```ts
+interface RuntimeField {
+  name: string;
+  type: RuntimeType; // 'long' | 'boolean' ...
+  script: {
+    source: string;
+  }
+}
+```
+
+#### Context object
+
+You can provide a context object to the runtime field editor. It has the following interface
+
+```ts
+interface Context {
+  /** An array of field name not allowed. You would probably provide an array of existing runtime fields
+   * to prevent the user creating a field with the same name.
+   */
+  namesNotAllowed?: string[];
+  /**
+   * An array of existing concrete fields. If the user gives a name to the runtime
+   * field that matches one of the concrete fields, a callout will be displayed
+   * to indicate that this runtime field will shadow the concrete field.
+   */
+  existingConcreteFields?: string[];
```

Comment:
nit; I'm not sure the word "concrete" adds much clarity for me here. I think just "existingFields" would be clearer + the comment which is very helpful

Reply:
But then how the consumer knows that this refers to concrete (mapped) fields and not runtime fields?